### ISSUE TITLE
PHP Warning: Division by zero

### ISSUE THREAD

**Issue by PedroAmorim** [andreskrey/readability.php#32]
Hello!

I have the warning:
`
PHP Warning: Division by zero ... andreskrey/readability.php/src/HTMLParser.php on line 1093
`
With the site:
[https://www.womenshealthmag.com/life/macys-black-friday-deals-2017](https://www.womenshealthmag.com/life/macys-black-friday-deals-2017)

Refer to lines in method **prepArticle**:

```php
      $h2 = $article->getElementsByTagName('h2');
        if ($h2->length === 1) {
           $lengthSimilarRate = (mb_strlen($h2->item(0)->textContent) - mb_strlen($this->metadata['title'])) / mb_strlen($this->metadata['title']);
```

It's caused by an empty metadata['title'], but I don't know why is empty.
I fix that with a `!empty(metadata['title'])` in the if statement, but I'm sure isn't the right way.

**Comment by andreskrey** [comment]
Interesting case. There's no title in the metadata because they are incorrectly closing a comment tag with a long dash instead of two dashes. Their source code looks like one giant comment blob. Here's the offending tag:

`<!-- End comScore Tag —> `
 
There's not much we can do in this case, but I pushed a change to avoid dividing by zero.

Thanks!

Fixed via https://github.com/andreskrey/readability.php/commit/acb1ff3b3065bcd3305a81c8b7d98f9d6aac3675
